uplink/storage/streams/store.go

```diff
@@ -223,10 +230,10 @@ func (s *streamStore) upload(ctx context.Context, path Path, pathCipher storj.Ci
 			}
 
 			streamInfo, err := proto.Marshal(&pb.StreamInfo{
-				NumberOfSegments: currentSegment + 1,
-				SegmentsSize:     s.segmentSize,
-				LastSegmentSize:  sizeReader.Size(),
-				Metadata:         metadata,
+				DeprecatedNumberOfSegments: currentSegment + 1,
```

Comment:
why do we add 1 to currentSegment to get DeprecatedNumberOfSegments?

Reply:
currentSegment is the index of the current (the last) segment, so the index + 1 is the number of segments from zero indexing (first segment is 0). so this is just the number of segments and we're filling the deprecated field